tests/sentry/notifications/test_helpers.py

```diff
@@ -0,0 +1,346 @@
+import types
+from urllib.parse import parse_qs, urlparse
+
+from sentry.models import NotificationSetting, Rule
+from sentry.models.notificationsettingoption import NotificationSettingOption
+from sentry.models.notificationsettingprovider import NotificationSettingProvider
+from sentry.notifications.helpers import (
+    collect_groups_by_project,
+    get_all_setting_options,
+    get_all_setting_providers,
+    get_scope_type,
+    get_setting_options_for_recipient,
+    get_settings_by_provider,
+    get_subscription_from_attributes,
+    get_values_by_provider_by_type,
+    user_has_any_provider_settings,
+    validate,
+)
+from sentry.notifications.notify import notification_providers
+from sentry.notifications.types import (
+    NotificationScopeEnum,
+    NotificationScopeType,
+    NotificationSettingEnum,
+    NotificationSettingOptionValues,
+    NotificationSettingsOptionEnum,
+    NotificationSettingTypes,
+)
+from sentry.notifications.utils import (
+    get_email_link_extra_params,
+    get_group_settings_link,
+    get_rules,
+)
+from sentry.testutils.cases import TestCase
+from sentry.types.integrations import ExternalProviderEnum, ExternalProviders
+
+
+def mock_event(*, transaction, data=None):
+    return types.SimpleNamespace(data=data or {}, transaction=transaction)
+
+
+def add_notification_setting_option(
+    scope_type,
+    scope_identifier,
+    type,
+    value,
+    user_id=None,
+    team_id=None,
+):
+    return NotificationSettingOption.objects.create(
+        scope_type=scope_type,
+        scope_identifier=scope_identifier,
+        type=type,
+        value=value,
+        user_id=user_id,
+        team_id=team_id,
+    )
+
+
+def add_notification_setting_provider(
+    scope_type,
+    scope_identifier,
+    provider,
+    type,
+    value,
+    user_id=None,
+    team_id=None,
+):
+    return NotificationSettingProvider.objects.create(
+        scope_type=scope_type,
+        scope_identifier=scope_identifier,
+        provider=provider,
+        type=type,
+        value=value,
+        user_id=user_id,
+        team_id=team_id,
+    )
+
+
+class NotificationHelpersTest(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        NotificationSetting.objects.update_settings(
+            ExternalProviders.SLACK,
+            NotificationSettingTypes.WORKFLOW,
+            NotificationSettingOptionValues.ALWAYS,
+            user_id=self.user.id,
+        )
+        NotificationSetting.objects.update_settings(
+            ExternalProviders.SLACK,
+            NotificationSettingTypes.DEPLOY,
+            NotificationSettingOptionValues.ALWAYS,
+            user_id=self.user.id,
+        )
+        NotificationSetting.objects.update_settings(
```

Comment:
@snigdhas do we need to generate `NotificationSetting` values?

Reply:
These are existing tests, split out from `tests/sentry/notifications/test_utils.py` so I'm leaving them unchanged. The new tests are in `NotificationSettingV2HelpersTest`